vms/platformvm/state/merkle_state.go

```diff
@@ -286,6 +286,31 @@ func newMerkleState(
 	}, nil
 }
 
+// Stores global state in a merkle trie. This means that each state corresponds
+// to a unique merkle root. Specifically, the following state is merkleized.
+// - Delegatee Rewards
+// - UTXOs
+// - Current Supply
+// - Subnet Creation Transactions
+// - Subnet Owners
+// - Subnet Transformation Transactions
+// - Chain Creation Transactions
+// - Chain time
+// - Last Accepted Block ID
+// - Current Staker Set
+// - Pending Staker Set
+//
+// Changing any of the above state will cause the merkle root to change.
+//
+// The following state is not merkleized:
+// - Database Initialization Status
+// - Blocks
+// - Block IDs
+// - Transactions (note some transactions are also stored merkleized)
+// - Uptimes
+// - Weight Diffs
+// - BLS Key Diffs
```

Comment:
Without this P-chain can't fully support Warp right after state sync. However we can find ways to download just enough blocks below state sync height to fill a few indexes. Also I am working on a block backfilling feature that may help with this.